Shell/Shell.cpp

```diff
@@ -1559,6 +1560,25 @@ void Shell::cache_path()
 
 void Shell::highlight(Line::Editor& editor) const
 {
+    auto resolve_path = [](auto& cwd, auto* host, auto& path) -> Line::Style {
+        LexicalPath lexical_path { path };
+        URL url;
+        if (lexical_path.is_absolute()) {
+            if (!Core::File::exists(lexical_path.string()))
```

Comment:
Could we avoid using lexical paths for accessing actual files? I'm not sure how, because the path in URL is lexical, but it's entirely possible that the file at `foo/../bar` exists, but `bar` doesn't. Perhaps get a `realpath` from the FS and use that for the URL?

Reply:
I did think of that, but felt that spamming syscalls at every entered character was just too much.
I'm not entirely sure if the edge-case is worth spamming syscalls, ideas?

aside, we should really have a decent way to join two paths reliably...